vm/qemu/qemu.go

```diff
@@ -350,30 +365,44 @@ func (inst *instance) boot() error {
 	args := []string{
 		"-m", strconv.Itoa(inst.cfg.Mem),
 		"-smp", strconv.Itoa(inst.cfg.CPU),
-		"-net", "nic" + inst.archConfig.NicModel,
-		"-net", fmt.Sprintf("user,host=%v,hostfwd=tcp::%v-:22", hostAddr, inst.port),
 		"-display", "none",
 		"-serial", "stdio",
 		"-no-reboot",
 	}
 	args = append(args, splitArgs(inst.cfg.QemuArgs, filepath.Join(inst.workdir, "template"))...)
+	if inst.archConfig.UseNewQemuNetOptions {
+		args = append(args,
+			"-device", "virtio-net-device,netdev=net0",
+			"-netdev", fmt.Sprintf("user,id=net0,host=%v,hostfwd=tcp::%v-:22", hostAddr, inst.port))
+	} else {
+		args = append(args,
+			"-net", "nic"+inst.archConfig.NicModel,
+			"-net", fmt.Sprintf("user,host=%v,hostfwd=tcp::%v-:22", hostAddr, inst.port))
+	}
 	if inst.image == "9p" {
 		args = append(args,
 			"-fsdev", "local,id=fsdev0,path=/,security_model=none,readonly",
 			"-device", "virtio-9p-pci,fsdev=fsdev0,mount_tag=/dev/root",
 		)
 	} else if inst.image != "" {
-		// inst.cfg.ImageDevice can contain spaces
-		imgline := strings.Split(inst.cfg.ImageDevice, " ")
-		imgline[0] = "-" + imgline[0]
-		if strings.HasSuffix(imgline[len(imgline)-1], "file=") {
-			imgline[len(imgline)-1] = imgline[len(imgline)-1] + inst.image
+		if inst.archConfig.UseNewQemuImageOptions {
+			args = append(args,
+				"-device", "virtio-blk-device,drive=hd0",
+				"-drive", fmt.Sprintf("file=%v,if=none,format=raw,id=hd0", inst.image),
+			)
 		} else {
-			imgline = append(imgline, inst.image)
-		}
-		args = append(args, imgline...)
-		if inst.cfg.Snapshot {
-			args = append(args, "-snapshot")
+			// inst.cfg.ImageDevice can contain spaces
+			imgline := strings.Split(inst.cfg.ImageDevice, " ")
+			imgline[0] = "-" + imgline[0]
+			if strings.HasSuffix(imgline[len(imgline)-1], "file=") {
+				imgline[len(imgline)-1] = imgline[len(imgline)-1] + inst.image
+			} else {
+				imgline = append(imgline, inst.image)
+			}
+			args = append(args, imgline...)
+			if inst.cfg.Snapshot {
+				args = append(args, "-snapshot")
```

Comment:
We need this part in both cases. Otherwise it will corrupt image and does not start with count>1.

Reply:
Oh, no, sorry. That's because I added -snapshot to qemu_args, but did not specify the machine.

Reply:
I think your first comment is right, we need this in both cases. Otherwise, I occasionally get:

```
2020/06/27 14:23:00 vm-0: crash: no output from test machine
2020/06/27 14:23:00 file /buildroot/output/images/rootfs.ext2 that syz-manager uses has been modified by an external program
this can lead to arbitrary syz-manager misbehavior
modification time has changed: 2020-06-27 14:09:31.200959653 +0200 CEST -> 2020-06-27 14:22:40.781257174 +0200 CEST
don't modify files that syz-manager uses. exiting to prevent harm
```